client.go

```diff
@@ -1459,7 +1459,7 @@ func (c *HostClient) acquireConn(reqTimeout time.Duration) (cc *clientConn, err
 		if c.connsCount < maxConns {
 			c.connsCount++
 			createConn = true
-			if !c.connsCleanerRun {
+			if !(c.connsCleanerRun || connectionClose) {
```

Comment:
```suggestion
			if !c.connsCleanerRun && !connectionClose {
```

Reply:
Just de morgana rule) I think readability depends on reader in this case) But ok, it's not a problem.